src/oemof/solph/_models.py

```diff
@@ -441,3 +444,46 @@ def relax_problem(self):
         relaxer._apply_to(self)
 
         return self
+
+
+class Results:
+    # TODO:
+    #   Defer attribute references not present as variables to
+    #   attributes of `model.solver_results` in order to make `Results`
+    #   instances returnable by `model.solve` and still be backwards
+    #   compatible.
+    def __init__(self, model: Model):
+        # TODO: Disambiguate colliding variable names.
+        self.variables = {
+            str(variable).split(".")[-1]: variable
+            for vardata in model.component_data_objects(Var)
+            for variable in [vardata.parent_component()]
+        }
```

Comment:
I think we could replace the dictionary comprehension by creating an empty dictionary and adding items afterwards. This way, it was possible to find name collisions.

```suggestion
        self.variables = {}
        for vardata in model.component_data_objects(Var):
            for variable in [vardata.parent_component()]:
                key = str(variable).split(".")[-1]
                if key not in self.variables:
                    self.variables[key] = variable
                else:
                    raise ValueError(
                        f"Variable name defined multiple times: {key}"
                        + f"(last time in '{variable}')"
                    )
```

Reply:
I am not a big fan of list/dict comprehension for more than one iteration loop (unless the operation is super obvious and simple), I like the for loop :). On the plus side it makes debugging  easier

Reply:
What is the usecase where the variable names are defined multiple times ?

Reply:
You can have the same variable name defined in multiple blocks. At the dev meeting, we decided to flatten this (e.g. "flow" instead of "SimpleFlowBlock.flow"). However, nobody prevents you from defining "MyOwnBlock.flow" as well, which then collides with the other variable.

Reply:
The change also revealed another problem: One variable was added thousands of times. (The current version avoids this, but still iterates over the same variable as often as before.)